pkg/handler3/handler_test.go

```diff
@@ -64,7 +64,7 @@ func TestRegisterOpenAPIVersionedService(t *testing.T) {
 	}
 
 	mux := http.NewServeMux()
-	o, err := NewOpenAPIService(nil)
```

Comment:
I checked the test suite for this and it does not seem to flex the mechanism for changing a groupversions content at all in the tests. Seeing as this PR pertains most to that functionality, do you think you can add some?